core/loadpoint_plan.go

```diff
@@ -118,13 +118,12 @@ func (lp *Loadpoint) plannerActive() (active bool) {
 		lp.log.TRACE.Printf("  slot from: %v to %v cost %.3f", slot.Start.Round(time.Second).Local(), slot.End.Round(time.Second).Local(), slot.Price)
 	}
 
-	activeSlot := planner.ActiveSlot(lp.clock, plan)
+	activeSlot := planner.SlotAt(lp.clock.Now(), plan)
 	active = !activeSlot.End.IsZero()
 
 	if active {
 		// ignore short plans if not already active
-		// TODO only ignore if the next adjacent slot is inactive, too
-		if slotRemaining := lp.clock.Until(activeSlot.End); !lp.planActive && slotRemaining < smallSlotDuration {
+		if slotRemaining := lp.clock.Until(activeSlot.End); !lp.planActive && slotRemaining < smallSlotDuration && !planner.SlotHasSuccessor(activeSlot, plan) {
```

Comment:
Sieht für mich auch gut aus. Also wenn man mit einem 5min Slot starten würde und er noch nicht läd, ist die Änderung das er den nach hinten an das Slot Ende schiebt 5min Läd und dann in einen vollen Slot übergeht. Finde ich gut.

Reply:
Genau so!